src/router.ts

```diff
@@ -203,13 +196,9 @@ const convertRouterSchemaToJSONSchemaStyle = <Schema extends ZodSchema>(
       Description: definition.description,
       // The JSONSchema types are very slightly different between packages. We just
       // trust that the interop will work fine, and use "as any" here.
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      Request: zodToJsonSchema(definition.request, {
-        $refStrategy: 'none',
-      }) as any,
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-      Response: zodToJsonSchema(definition.response, {
-        $refStrategy: 'none',
+      Request: z.toJSONSchema(definition.request, { target: 'draft-7' }) as any,
```

Comment:
`zod` 4 now has native JSON Schema support, so we can use that instead of having to use the `zod-to-json-schema` package, which is not currently being developed on.